base/irrationals.jl

```diff
@@ -221,6 +221,16 @@ end
 log(::Irrational{:e}) = 1 # use 1 to correctly promote expressions like log(x)/log(e)
 log(::Irrational{:e}, x::Number) = log(x)
 
+# These methods should go to special/trig.jl, but Irrational is not yet defined when the
+# file is loaded.
+sinpi(x::Irrational) = sinpi(float(x))
+cospi(x::Irrational) = cospi(float(x))
+sinc(x::Irrational) = sinc(float(x))
+sinc(z::Complex{T}) where {T<:Irrational} = sinc(float(z))
```

Comment:
Could this signature simplify to `sinc(z::Complex{<:Irrational})`, and likewise for the similar `cosc` definition below?